hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -73,6 +74,7 @@
   private String buildDate;
   private HddsProtos.NodeOperationalState persistedOpState;
   private long persistedOpStateExpiryEpochSec = 0;
+  private final List<String> features = new ArrayList<>();
```

Comment:
I might have missed something:

Is there a JIRA or design doc that says what features in Datanode are?

Reply:
This is just a way to help distinguish newly initialized datanodes from existing ones being restarted with new code version.  No further features planned currently.